WordPressKit/Jetpack Scan/JetpackScanThreat.swift

```diff
@@ -0,0 +1,253 @@
+        case core
+        case file
+        case plugin
+        case theme
+        case database
+
+        case unknown
+        static let unknownCase: Self = .unknown
+
+        init(threat: JetpackScanThreat) {
+            // Logic used from https://github.com/Automattic/wp-calypso/blob/5a6b257ad97b361fa6f6a6e496cbfc0ef952b921/client/components/jetpack/threat-item/utils.ts#L11
+            if threat.diff != nil {
+                self = .core
+            } else if threat.context != nil {
+                self =  .file
+            } else if let ext = threat.extension {
+                self = ThreatType(rawValue: ext.type.rawValue)
+            } else if threat.rows != nil || threat.signature.contains(Constants.databaseSignature) {
+                self = .database
+            } else {
+                self = .unknown
+            }
+        }
+
+        private struct Constants {
+            static let databaseSignature = "Suspicious.Links"
+        }
+    }
+
+    /// The ID of the threat
+    public let id: Int
+
+    /// The name of the threat signature
+    public let signature: String
+
+    /// The description of the threat signature
+    public let description: String
+
+    /// The date the threat was first detected
+    public let firstDetected: Date
+
+    /// Whether the threat can be automatically fixed
+    public let fixable: JetpackScanThreatFixer?
+
+    /// The filename
+    public let fileName: String?
+
+    /// The status of the threat (fixed, ignored, current)
+    public let status: ThreatStatus?
+
+    /// The date the threat was fixed on
+    public let fixedOn: Date?
+
+    /// More information if the threat is a extension type (plugin or theme)
+    public let `extension`: JetpackThreatExtension?
+
+    /// The type of threat this is
+    private(set) var type: ThreatType = .unknown
+
+    /// If this is a file based threat this will provide code context to be displayed
+    /// Context example:
+    /// 3: start test
+    /// 4: VIRUS_SIG
+    /// 5: end test
+    /// marks: 4: (0, 9)
+    public let context: JetpackThreatContext?
+
+    // Core modification threats will contain a git diff string
+    public let diff: String?
+
+    // Database threats will contain row information
+    public let rows: [String: Any]?
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        id = try container.decode(Int.self, forKey: .id)
+        signature = try container.decode(String.self, forKey: .signature)
+        fileName = try container.decodeIfPresent(String.self, forKey: .fileName)
+        description = try container.decode(String.self, forKey: .description)
+        firstDetected = try container.decode(Date.self, forKey: .firstDetected)
+        fixedOn = try container.decodeIfPresent(Date.self, forKey: .fixedOn)
+        fixable = try? container.decodeIfPresent(JetpackScanThreatFixer.self, forKey: .fixable) ?? nil
```

Comment:
```suggestion
        fixable = try? container.decodeIfPresent(JetpackScanThreatFixer.self, forKey: .fixable)
```

Reply:
I left that in because the value can either be `false` or an `object` so I'm using `try?` to allow me to default to nil and prevent a decoding failure because the value isn't an object.